src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -59,30 +58,40 @@ private RazorLanguageServer(ILanguageServer innerServer)
 
         public static Task<RazorLanguageServer> CreateAsync(Stream input, Stream output, Trace trace, Action<RazorLanguageServerBuilder> configure = null)
         {
-            Serializer.Instance.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
+            // Serializer.Instance.Settings.Converters.Add(SemanticTokensOrSemanticTokensEditsConverter.Instance);
             Serializer.Instance.JsonSerializer.Converters.RegisterRazorConverters();
 
             // Custom ClientCapabilities deserializer to extract experimental capabilities
             Serializer.Instance.JsonSerializer.Converters.Add(ExtendableClientCapabilitiesJsonConverter.Instance);
 
             ILanguageServer server = null;
+            var logLevel = trace switch
+            {
+                Trace.Messages => LogLevel.Information,
+                Trace.Off => LogLevel.None,
+                Trace.Verbose => LogLevel.Trace,
+                _ => default,
+            };
+
             server = OmniSharp.Extensions.LanguageServer.Server.LanguageServer.PreInit(options =>
                 options
                     .WithInput(input)
                     .WithOutput(output)
+                    .WithContentModifiedSupport(false)
+                    .WithSerializer(Serializer.Instance)
```

Comment:
ooo this is dope af. We may want to consider removing our usage of the Serializer.Instance type so we aren't polluting a global static